applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing_matrix_free.h

```diff
@@ -85,12 +85,14 @@ class MapperVertexMorphingMatrixFree : public Mapper
     ///@{
 
     /// Default constructor.
-    MapperVertexMorphingMatrixFree( ModelPart& designSurface, Parameters MapperSettings )
-        : mrDesignSurface( designSurface ),
-          mNumberOfDesignVariables( designSurface.Nodes().size() ),
-          mFilterType( MapperSettings["filter_function_type"].GetString() ),
+    MapperVertexMorphingMatrixFree( ModelPart& rOriginMdpa, ModelPart& rDestinationMdpa, Parameters MapperSettings )
+        : mrOriginMdpa( rOriginMdpa ),
+          mrDestinationMdpa( rDestinationMdpa ),
```

Comment:
i think we could easily remove these variables: `mOriginNodeNumber`, `mDestinationNodeNumber` - its kind of duplicated information, because we already store the modelpart.